test/libsolidity/ASTPropertyTest.cpp

```diff
@@ -150,17 +150,22 @@ void ASTPropertyTest::extractTestsFromAST(Json::Value const& _astJson)
 					);
 					m_tests[testId].property = testedProperty;
 
-					soltestAssert(node.isMember("nodeType"));
-					std::optional<Json::Value> propertyNode = jsonValueByPath(node, testedProperty);
+					soltestAssert(node.contains("nodeType"));
+					std::optional<Json> propertyNode = jsonValueByPath(node, testedProperty);
 					soltestAssert(
 						propertyNode.has_value(),
-						node["nodeType"].asString() + " node does not have a property named \""s + testedProperty + "\""
+						node["nodeType"].get<std::string>() + " node does not have a property named \""s + testedProperty + "\""
 					);
 					soltestAssert(
-						!propertyNode->isObject() && !propertyNode->isArray(),
+						!propertyNode->is_object() && !propertyNode->is_array(),
 						"Property \"" + testedProperty + "\" is an object or an array."
 					);
-					m_tests[testId].obtainedValue = propertyNode->asString();
```

Comment:
Was this a bug where we didn't handle boolean before, or did jsoncpp just implicitly convert bools to strings?

Reply:
Yes, looks like.

Reply:
It's just testing, precise behaviour is not too important as long as it still tests what it should